src/kernel/src/regmgr/command.rs

```diff
@@ -0,0 +1,30 @@
+use super::RegError;
+
+#[repr(u32)]
+pub(super) enum RegMgrCommand<'a> {
+    SetInt(&'a SetIntArg) = 0x18,
+    Unk1(&'a Unk1Arg) = 0x19,
+}
+impl RegMgrCommand<'_> {
+    pub fn try_from_raw_parts(cmd: u32, arg: *const u8) -> Result<Self, RegError> {
```

Comment:
This method should be unsafe and bind a lifetime to `arg` otherwise it is possible for the returned object to outlive the `arg`.

Reply:
Actually, I think it will be better to make it take a SysArg directly